--- a/src/rshrcmd.c	2009-07-03 15:06:04.000000000 +1000
+++ b/src/rshrcmd.c	2009-07-03 15:08:34.000000000 +1000
@@ -72,7 +72,7 @@ rshrcmd(ahost, port, luser, ruser, cmd, 
 		/* child. we use sp[1] to be stdin/stdout, and close
 		   sp[0]. */
 		(void) close(sp[0]);
-		if (dup2(sp[1], 0) < 0 || dup2(0,1) < 0 || dup2(0, 2) < 0) {
+		if (dup2(sp[1], 0) < 0 || dup2(0,1) < 0) {
 			error("dup2 failed: %s.", SYSERR);
 			_exit(255);
 		}
